domain/application/service/application.go

```diff
@@ -1028,34 +1028,22 @@ func (s *Service) deleteApplication(ctx domain.AtomicContext, name string) ([]fu
 // if the application doesn't exist.
 func (s *Service) DestroyApplication(ctx context.Context, appName string) error {
 	// For now, all we do is advance the application's life to Dying.
-	err := s.st.RunAtomic(ctx, func(ctx domain.AtomicContext) error {
-		appID, err := s.st.GetApplicationID(ctx, appName)
-		if errors.Is(err, applicationerrors.ApplicationNotFound) {
-			return nil
-		}
-		if err != nil {
-			return errors.Trace(err)
-		}
-		return s.st.SetApplicationLife(ctx, appID, life.Dying)
-	})
-	return errors.Annotatef(err, "destroying application %q", appName)
+	err := s.st.SetApplicationLife(ctx, appName, life.Dying)
+	if err != nil {
+		return internalerrors.Errorf("destroying application %q: %w", appName, err)
+	}
+	return nil
 }
 
 // EnsureApplicationDead is called by the cleanup worker if a mongo
 // destroy operation sets the application to dead.
 // TODO(units): remove when everything is in dqlite.
 func (s *Service) EnsureApplicationDead(ctx context.Context, appName string) error {
```

Comment:
Shouldn't this just be called, `MarkApplicationDead`?

Reply:
That might be better, although I don't think it's necessary because `EnsureApplicationDead` isn't misleading